Mathlib/Tactic/ComputeDegree.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Data.Polynomial.Degree.Definitions
+
+/-!
+###  Simple lemmas about `natDegree` and `degree`
+
+The lemmas in this section deduce inequalities of the form `natDegree f ≤ d` and `degree f ≤ d`,
+using inequalities of the same shape.
+This allows a recursive application of the `compute_degree_le` tactic on a goal,
+and on all the resulting subgoals.
+-/
+
+open Polynomial
+
+namespace Mathlib.Tactic.ComputeDegree
+
+section mylemmas
+
+variable {R : Type _}
+
+section semiring
+variable [Semiring R]
+
+theorem add {a b : Nat} {f g : R[X]} (hf : natDegree f ≤ a) (hg : natDegree g ≤ b) :
```

Comment:
IMHO, this can go near `Polynomial.natDegree_add_le`, e.g., under name `Polynomial.natDegree_add_le_of_le`. See `norm_add_le_of_le` for an example of this pattern already in the library.

Same for most of the lemmas below.